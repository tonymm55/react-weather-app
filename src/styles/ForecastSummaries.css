.forecast-summaries {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-top: 5rem;
}

.forecast-summary button {
    display: inline-block;
  padding: .2rem .2rem;
  background-color: black;
  color: white;
  font-size: 1.rem;
  text-align: center;
  text-decoration: none;
  transition: all 0.3s ease;
}

.forecast-summary button:active {
    background-color: white;
    color: black;
  }


