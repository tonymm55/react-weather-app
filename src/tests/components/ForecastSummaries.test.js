import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        description: "Stub description 1",
        icon: 800,
        temperature: {
          max: 22,
          min: 12,
        },
      },
      {
        date: 2222222,
        description: "Stub description 2",
        icon: 600,
        temperature: {
          max: 24,
          min: 13,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps.forecasts} />,
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});

describe("renders correct values for props", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description",
      icon: 800,
      temperature: {
        max: 22,
        min: 12,
      },
    },
  ];

  it("renders correctly", () => {
    const { getByText } = render(<ForecastSummaries forecasts={validProps} />);
    expect(getByText("Stub description")).toBeInTheDocument();
  });
});

describe("ForecastSummaries", () => {
  const forecasts = [
    {
      date: "2023-03-25",
      description: "Sunny",
      icon: 800,
      temperature: {
        max: 20,
        min: 10,
      },
    },
    {
      date: "2023-03-26",
      description: "Cloudy",
      icon: 600,
      temperature: {
        max: 18,
        min: 8,
      },
    },
  ];

  it("renders the correct number of ForecastSummary components", () => {
    render(<ForecastSummaries forecasts={forecasts} />);
    const forecastSummaryElements = screen.getAllByTestId("forecast-summary");
    expect(forecastSummaryElements.length).toEqual(2);
  });
});
