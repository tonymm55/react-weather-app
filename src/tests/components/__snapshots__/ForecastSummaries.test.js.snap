// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="icon"
          src="https://openweathermap.org/img/wn/01d@2x.png"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
      <button
        type="button"
      >
        Click for details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="icon"
          src="https://openweathermap.org/img/wn/13d@2x.png"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
      <button
        type="button"
      >
        Click for details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
